R/merge.R

```diff
@@ -11,9 +11,21 @@ merge.data.table = function(x, y, by = NULL, by.x = NULL, by.y = NULL, all = FAL
       by = key(x)
     }
   }
-  if ((x0 <- length(x)==0L) | (y0 <- length(y)==0L)) warning("You are trying to join data.tables where ", if(x0 & y0) "'x' and 'y' arguments are" else if(x0 & !y0) "'x' argument is" else if(!x0 & y0) "'y' argument is", " 0 columns data.table.")
```

Comment:
this was implicitly relying on `|` because `y0 <-` won't be evaluated if `x0`. Not sure that's a good reason to use `|`, so I rewrote as below. Looks cleaner IMO.